### ISSUE TITLE
default typemap & autodoc python regression from 3.0.3

### ISSUE THREAD

**Issue by xantares** [swig/swig#330]
There may be another python regression since 3.0.3.
consider wrapping this from c++:
```
%feature("autodoc", "somedocstring") myopen;
%typemap(default) const char * connstring {$1 = 0;}
void myopen(int a, const char * connstring);
```

then when calling without the connstring argument:
```
ret = example.myopen(4)
```
raises
```
TypeError: myopen() takes exactly 2 arguments (1 given)
```
If I remove the autodoc feature the default argument is understood.
Ok with 3.0.2, bad in 3.0.3, 3.0.5 & master.
Same with module global %feature("autodoc","1");

**Comment by vadz** [comment]
I can confirm that this can be reproduced with
```
diff --git a/Examples/test-suite/autodoc.i b/Examples/test-suite/autodoc.i
index d858997..8e05681 100644
--- a/Examples/test-suite/autodoc.i
+++ b/Examples/test-suite/autodoc.i
@@ -29,12 +29,16 @@

 %feature("autodoc","just a string") A::funk; // names

+%typemap(default) int y {$1 = 0;}
+
 %inline {

   enum Hola {
     hi, hello
   };
-
+
+  int func_with_default_arg(int x, int y) { return x+y; }
+
   struct A {
     A(int a, short b, Hola h) {}
     int funk(int a) { return a; }
diff --git a/Examples/test-suite/python/autodoc_runme.py b/Examples/test-suite/python/autodoc_runme.py
index 9b493bb..9df9c44 100644
--- a/Examples/test-suite/python/autodoc_runme.py
+++ b/Examples/test-suite/python/autodoc_runme.py
@@ -20,6 +20,10 @@ skip = True # skip builtin check - the autodoc is missing, but it probably shoul

 check(A.__doc__, "Proxy of C++ A class", "::A")
 check(A.funk.__doc__, "just a string")
+if func_with_default_arg(17) != 17:
+  raise RuntimeError
+if func_with_default_arg(3, 4) != 7:
+  raise RuntimeError
 check(A.func0.__doc__,
 "func0(self, arg2, hello) -> int",
 "func0(arg2, hello) -> int")
```

I guess we need to check for the presence of `default` typemap for any of the function arguments and generate `*args` signature for it in this case, but unfortunately I don't know how to do this yet. Any hints would be welcome...

**Comment by vadz** [comment]
FWIW it seems to work only accidentally without `autodoc`, due to this strange construction in SWIG-generated code:
```python
def func(x, y):
    return _module.func(x, y)
func = _module.func
```
where the last line is present only if `autodoc` is off.

Also notice that everything seems to work fine if `-builtin` is used which provides at least a workaround, if not a solution.

**Comment by xantares** [comment]
still no one on this ?

**Comment by xantares** [comment]
@vadz, @wsfulton any news on this ?

**Comment by vadz** [comment]
This should be fixed by #389, please test.

**Comment by vadz** [comment]
Should be fixed by 5569d91bd02432eef4fb7ef2c2ed0df2f51c8472.

**Comment by xantares** [comment]
@vadz thanks a lot!
